resources/src/asset.rs

```diff
@@ -0,0 +1,99 @@
+use utils;
+/// Permissions around who can own an asset and whether or
+/// not the asset issuer can freeze the asset.
+#[derive(Serialize, Deserialize, Debug)]
+struct Flag {
+    auth_required: bool,
+    auth_revocable: bool,
+}
+
+/// Assets are the units that are traded on the Stellar Network.
+/// An asset consists of an type, code, and issuer.
+/// Any asset can be traded for any other asset.
+/// https://www.stellar.org/developers/horizon/reference/resources/asset.html
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Asset {
+    asset_type: String,
+    asset_code: String,
+    asset_issuer: String,
+    #[serde(deserialize_with = "utils::deserialize_from_str")] amount: f64,
+    num_accounts: u32,
+    flags: Flag,
+}
+
+impl Asset {
+    ///   The type of this asset: “credit_alphanum4”, or “credit_alphanum12”.
+    ///   Returns a slice that lives as long as the asset does.
+    pub fn asset_type<'a>(&'a self) -> &'a str {
+        &(self.asset_type)
+    }
+
+    /// The code of this asset.
+    /// Returns a slice that lives as long as the asset does.
+    pub fn asset_code<'a>(&'a self) -> &'a str {
+        &(self.asset_code)
+    }
+
+    /// The issuer of this asset.  This corresponds to the id of an account.
+    /// Returns a slice that lives as long as the asset does.
+    pub fn asset_issuer<'a>(&'a self) -> &'a str {
+        &(self.asset_issuer)
+    }
+
+    /// The number of units of credit issued for this asset.
+    /// This number is scaled by 10 million to display the number if the format a
+    /// user would expect it in
+    /// https://www.stellar.org/developers/guides/concepts/assets.html
+    /// Returns a signed 64-bit integer.
+    pub fn amount(&self) -> i64 {
+        (self.amount * 10_000_000.0) as i64
```

Comment:
Two things come to mind here:

1. We don't want to incur a multiplication every time we look at amount.
2. Multiplying large floating points can lead to loss of precision which would not be good here.

I'd recommend implementing a custom deserializer to convert to amount. One where we determine the offset of the decimal point, sub out the decimal from a string, then convert to a i64, and multiply by the orders of maginitude between the offset at 10 million.

Definitely slightly tricky code but the result should retain the precision that stellar has.